pennylane/transforms/split_non_commuting.py

```diff
@@ -47,7 +97,10 @@ def shot_vector_postprocessing(results):
 
 @transform
 def split_non_commuting(
-    tape: QuantumScript, grouping_strategy: str | None = "default"
+    tape: QuantumScript,
+    grouping_strategy: str | None = "default",
+    shot_distribution: ShotDistributionFunction | str = None,
+    seed: int = None,
```

Comment:
```suggestion
    seed: None | int | np.random.Generator = None,
```
Typing a seed can be tricky, but I think this covers the major cases.